runtime-light/allocator/runtime-light-allocator.cpp

```diff
@@ -54,50 +54,51 @@ void RuntimeAllocator::free() {
 void *RuntimeAllocator::alloc_script_memory(size_t size) noexcept {
   php_assert(size);
   if (unlikely(!is_script_allocator_available())) {
-    return get_platform_context()->allocator.alloc(size);
+    return alloc_global_memory(size);
   }
 
   ComponentState &rt_ctx = *get_component_context();
 
   void *ptr = rt_ctx.runtime_allocator.memory_resource.allocate(size);
-  if (ptr == nullptr) {
-    request_extra_memory(size);
+  size_t previous_multiplier = 1;
+  while (ptr != nullptr) {
```

Comment:
Shouldn't we have a fallback here? What if `memory_resource.allocate` never returns a non-`nullptr` value?